proto/services/tasks_svc/v1/properties_svc.proto

```diff
@@ -0,0 +1,355 @@
+syntax = "proto3";
+
+package proto.services.tasks_svc.v1;
+
+option go_package = "gen/proto/services/tasks-svc";
+
+enum SubjectType {
+  SUBJECT_TYPE_UNSPECIFIED = 0;
+  SUBJECT_TYPE_PATIENT = 1;
+}
+
+enum FieldType {
+  FIELD_TYPE_UNSPECIFIED = 0;
+  FIELD_TYPE_TEXT = 1;
+  FIELD_TYPE_NUMBER = 2;
+  FIELD_TYPE_CHECKBOX = 3;
+  FIELD_TYPE_DATE = 4;
+  FIELD_TYPE_DATE_TIME = 5;
+  FIELD_TYPE_SELECT = 6;
+}
+
+// TODO: View"Context" is too general of a name for this in my opinion
+enum ViewContext {
+  VIEW_CONTEXT_UNSPECIFIED = 0;
+  VIEW_CONTEXT_GLOBAL = 1;
+  // TODO TBD: VIEW_CONTEXT_ORGANIZATION = 2;
+  VIEW_CONTEXT_WARD = 2;
+}
+
+/***************************************************/
+/********* Work with the Properties Schema *********/
+/***************************************************/
+
+//
+// CreatePropertySet
+//
+
+message CreatePropertySetRequest {
+  string name = 1;
+
+  repeated string properties = 3; // @gotags: validate:"dive,uuid4"
+}
+
+message CreatePropertySetResponse {
+  string property_set_id = 1;
+}
+
+//
+// UpdatePropertySet
+//
+
+
+message UpdatePropertySetRequest {
+  string set_id = 1; // @gotags: validate:"uuid4"
+  optional string name = 2; // TODO: add validation
+
+  repeated string added_properties = 3; // @gotags: validate:"dive,uuid4"
+  repeated string removed_properties = 4; // @gotags: validate:"dive,uuid4"
+}
+
+message UpdatePropertySetResponse {}
+
+//
+// UpdatePropertySet
+//
+
+
+message DeletePropertySetRequest {
+  string set_id = 1; // @gotags: validate:"uuid4"
+}
+
+message DeletePropertySetResponse {}
+
+//
+// CreateProperty
+//
+
+message CreatePropertyRequest {
+  ViewContext context = 1;
+  SubjectType subject_type = 2; // TODO: add validation here (not-unspecified check)
+  FieldType field_type = 3; // TODO: add validation here (not-unspecified check)
+
+  string name = 4; // TODO: add validation (at least non-empty checks)
+  optional string description = 5; // TODO: add validation, if needed
+
+  // ID of set this Property should belong to
+  optional string set_id = 6;
+
+  optional bool always_include_for_current_context = 7;
```

Comment:
In order to be able to do this, we need to get the context (above, proto value 1). For example "Ward".

Reply:
I think we need this, else the frontend would need to query the state of this button manually for every property:

![image](https://github.com/helpwave/services/assets/22842759/d54338d0-ba95-4776-9570-a3894db90057)

Reply:
Okay, looks good. As mentioned on Slack, we need to talk about the PatientViews separately.